python-package/brainflow/data_filter.py

```diff
@@ -609,6 +621,44 @@ def perform_wavelet_denoising(cls, data: NDArray[Float64], wavelet: str, decompo
         if res != BrainflowExitCodes.STATUS_OK.value:
             raise BrainFlowError('unable to denoise data', res)
 
+    @classmethod
+    def get_csp(cls, data: NDArray[Float64], labels: NDArray[Float64]) -> Tuple:
+        """calculate common spatial filters
+
+        :param data: [epochs x channels x times]-shaped 3D array of data for two classes
+        :type data: NDArray[Float64]
+        :param labels: n_epochs-length 1D array of zeros and ones that assigns class labels for each epoch. Zero corresponds to the first class
+        :type labels: NDArray[Int64] 
+        :return: [channels x channels]-shaped 2D array of filters and [channels]-length 1D array of corresponding eigenvalues
+        :rtype: Tuple
+        """
+        if not (len(labels.shape) == 1):
+            raise BrainFlowError('Invalid shape of array <labels>', BrainflowExitCodes.INVALID_ARGUMENTS_ERROR.value)
+        if not (len(labels) == data.shape[0]):
+            raise BrainFlowError('Invalid number of elements in array <labels>', BrainflowExitCodes.INVALID_ARGUMENTS_ERROR.value)
+        
+        n_epochs, n_channels, n_times = data.shape
+
+        temp_data1d = numpy.zeros((int(n_epochs * n_channels * n_times))).astype(numpy.float64)
+        for e in range(n_epochs):
+            for c in range(n_channels):
+                for t in range(n_times):
+                    temp_data1d[e * n_channels * n_times + c * n_times + t] = data[e, c, t]
```

Comment:
I think in slack you mentioned that you get rid off such loop in python, did you forget to push this change?